staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/encryptionconfig_test.go

```diff
@@ -0,0 +1,193 @@
+	// Transforms data using one of them, and tries to untransform using both of them.
+	// Repeats this for both the possible combinations.
+
+	identityFirstTransformerOverrides, err := ParseEncryptionConfiguration(strings.NewReader(correctConfigWithIdentityFirst))
+	if err != nil {
+		t.Fatalf("error while parsing configuration file: %s.\nThe file was:\n%s", err, correctConfigWithIdentityFirst)
+	}
+
+	aesFirstTransformerOverrides, err := ParseEncryptionConfiguration(strings.NewReader(correctConfigWithAesFirst))
+	if err != nil {
+		t.Fatalf("error while parsing configuration file: %s.\nThe file was:\n%s", err, correctConfigWithAesFirst)
+	}
+
+	// Pick the transformer for any of the returned resources.
+	identityFirstTransformer := identityFirstTransformerOverrides[schema.ParseGroupResource("secrets")]
+	aesFirstTransformer := aesFirstTransformerOverrides[schema.ParseGroupResource("secrets")]
+
+	context := value.DefaultContext([]byte(sampleContextText))
+	originalText := []byte(sampleText)
+
+	// Transform using aes-first.
+	// Untransform using both identity-first and aes-first.
+	aesTransformedData, err := aesFirstTransformer.TransformToStorage(originalText, context)
+	if err != nil {
+		t.Fatalf("error while transforming data to storage using AES transformer: %s", err)
+	}
+
+	aesUntransformedData, stale, err := aesFirstTransformer.TransformFromStorage(aesTransformedData, context)
+	if err != nil {
+		t.Fatalf("error while transforming data written by AES, from storage using AES transformer: %s", err)
+	}
+	if stale != false {
+		t.Fatalf("wrong stale information on using transformer with AES first, on data encrypted by identity. Should be false for AES transformer reads")
+	}
+
+	identityUntransformedData, stale, err := identityFirstTransformer.TransformFromStorage(aesTransformedData, context)
+	if err != nil {
+		t.Fatalf("error while transforming data written by AES, from storage, using identity: %s", err)
+	}
+	if stale != true {
+		t.Fatalf("wrong stale information on using transformer with AES first, on data encrypted by aes. Should be true for identity transformer reads")
+	}
+
+	if bytes.Compare(aesUntransformedData, originalText) != 0 {
+		t.Fatalf("aes-first transformer transformed data (written by aes-first) incorrectly. Expected: %v, got %v", originalText, aesUntransformedData)
+	}
+
+	if bytes.Compare(identityUntransformedData, originalText) != 0 {
+		t.Fatalf("identity-first transformer transformed data (written by aes-first) incorrectly. Expected: %v, got %v", originalText, aesUntransformedData)
+	}
+
+	// Transform using identity-first.
+	// Untransform using both identity-first and aes-first.
+	identityTransformedData, err := identityFirstTransformer.TransformToStorage(originalText, context)
+	if err != nil {
+		t.Fatalf("error while transforming data to storage using AES transformer: %s", err)
+	}
+
+	aesUntransformedData, stale, err = aesFirstTransformer.TransformFromStorage(identityTransformedData, context)
```

Comment:
This cries out for a table driven test - might be better to restructure in that form.  Or at least shorter error messages :)